packages/api/src/beacon/routes/beacon/index.ts

```diff
@@ -28,51 +26,41 @@ export type {
   StateId,
   ValidatorId,
   ValidatorStatus,
-  ValidatorFilters,
-  CommitteesFilters,
   FinalityCheckpoints,
   ValidatorResponse,
   ValidatorBalance,
   EpochCommitteeResponse,
   EpochSyncCommitteeResponse,
 } from "./state.js";
 
-export type Api = block.Api &
-  pool.Api &
-  state.Api &
-  rewards.Api & {
-    getGenesis(): Promise<ApiClientResponse<{[HttpStatusCode.OK]: {data: phase0.Genesis}}>>;
+export type Endpoints = block.Endpoints &
+  pool.Endpoints &
+  state.Endpoints &
+  rewards.Endpoints & {
+    getGenesis: Endpoint<
+      //
+      "GET",
+      EmptyArgs,
+      EmptyRequest,
+      phase0.Genesis,
+      EmptyMeta
+    >;
   };
 
-export const routesData: RoutesData<Api> = {
-  getGenesis: {url: "/eth/v1/beacon/genesis", method: "GET"},
-  ...block.routesData,
-  ...pool.routesData,
-  ...state.routesData,
-  ...rewards.routesData,
-};
-
-export type ReqTypes = {
-  [K in keyof ReturnType<typeof getReqSerializers>]: ReturnType<ReturnType<typeof getReqSerializers>[K]["writeReq"]>;
-};
-
-// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function getReqSerializers(config: ChainForkConfig) {
-  return {
-    getGenesis: reqEmpty,
-    ...block.getReqSerializers(config),
-    ...pool.getReqSerializers(),
-    ...state.getReqSerializers(),
-    ...rewards.getReqSerializers(),
-  };
-}
-
-export function getReturnTypes(): ReturnTypes<Api> {
+export function definitions(config: ChainForkConfig): RouteDefinitions<Endpoints> {
```

Comment:
As mentioned in one comment earlier, more clear to use `routes` or `routesDefinitions`. 

The word `definition` meant `the act of [defining] or of making something definite`. Unless we use another noun with it it's confusing what it is defining. 

```suggestion
export function definitions(config: ChainForkConfig): RouteDefinitions<Endpoints> {
```

Reply:
We should definitely avoid the usage of the term "routes" here because this is what we refer to what will be added to fastify as a route which is different from the route definitions here
https://github.com/ChainSafe/lodestar/blob/02351960ab0a2393230ed6df93643576179673dc/packages/api/src/utils/server/route.ts#L38-L42

> The word definition meant the act of [defining] or of making something definite. Unless we use another noun with it it's confusing what it is defining.

You could say that about almost everything if you look at it with 0 context whatsoever but these "definitions" are literally in the routes directory and are typed as `RouteDefinitions`, and since this name is not used other than wiring things up, I don't see a renaming helpful here other than increasing verbosity.